scripts/release/atomic_pipeline.py

```diff
@@ -0,0 +1,511 @@
+        "mongodb_tools_url_ubi": mongodb_tools_url_ubi,
+        "mongodb_agent_url_ubi": mongodb_agent_url_ubi,
+        "quay_registry": build_configuration.base_registry,
+    }
+
+    build_image(
+        image_name="mongodb-agent-ubi",
+        dockerfile_path="docker/mongodb-agent/Dockerfile",
+        build_configuration=build_configuration_copy,
+        extra_args=args,
+    )
+
+
+def build_agent_default_case(build_configuration: BuildConfiguration):
+    """
+    Build the agent only for the latest operator for patches and operator releases.
+
+    """
+    release = load_release_file()
+
+    # We need to release [all agents x latest operator] on operator releases
+    if build_configuration.scenario == BuildScenario.RELEASE:
+        agent_versions_to_build = gather_all_supported_agent_versions(release)
+    # We only need [latest agents (for each OM major version and for CM) x patch ID] for patches
+    else:
+        agent_versions_to_build = gather_latest_agent_versions(release)
+
+    logger.info(
+        f"Building Agent versions: {agent_versions_to_build} for Operator versions: {build_configuration.version}"
+    )
+
+    tasks_queue = Queue()
+    max_workers = 1
+    if build_configuration.parallel:
+        max_workers = None
+        if build_configuration.parallel_factor > 0:
+            max_workers = build_configuration.parallel_factor
+    with ProcessPoolExecutor(max_workers=max_workers) as executor:
+        logger.info(f"Running with factor of {max_workers}")
+        logger.info(f"======= Agent versions to build {agent_versions_to_build} =======")
+        for idx, agent_version in enumerate(agent_versions_to_build):
+            # We don't need to keep create and push the same image on every build.
+            # It is enough to create and push the non-operator suffixed images only during releases to ecr and quay.
+            logger.info(f"======= Building Agent {agent_version} ({idx}/{len(agent_versions_to_build)})")
+            _build_agent_operator(
+                agent_version,
+                build_configuration,
+                executor,
+                build_configuration.version,
+                tasks_queue,
+            )
+
+    queue_exception_handling(tasks_queue)
+
+
+def queue_exception_handling(tasks_queue):
+    exceptions_found = False
+    for task in tasks_queue.queue:
+        if task.exception() is not None:
+            exceptions_found = True
+            logger.fatal(f"The following exception has been found when building: {task.exception()}")
+    if exceptions_found:
+        raise Exception(
+            f"Exception(s) found when processing Agent images. \nSee also previous logs for more info\nFailing the build"
+        )
+
+
+def _build_agent_operator(
+    agent_version: Tuple[str, str],
+    build_configuration: BuildConfiguration,
+    executor: ProcessPoolExecutor,
+    operator_version: str,
+    tasks_queue: Queue,
+):
+    agent_distro = "rhel9_x86_64"
```

Comment:
you removed the the multiarch agent support with atomic

Reply:
Yes that's on purpose, Lucian is aware of that and these args computation should be done inside Dockerfiles.
I think the diagram in the TD is the best way to understand it